lib/rules/max-lines-per-function.js

```diff
@@ -133,6 +133,11 @@ module.exports = {
          * @private
          */
         function processFunction(node) {
+
+            // attempting to process a FunctionExpression that's already been processed by a MethodDefinition or Property above it - so ignore
+            if (node.type === "FunctionExpression" && node.parent && (node.parent.type === "MethodDefinition" || node.parent.type === "Property")) {
```

Comment:
Because computed property name can have other functions, we need to check `node === node.parent.value` if the parent node is a `Property` or a `MethodDefinition`.

And we need some additional checks if the parent node is a `Property`.

- if `parent.kind` is `"get"`/`"set"` then it's a getter/setter. (e.g., `get foo() {}`)
- if `parent.method` is `true` then it's a method shorthand. (e.g., `foo() {}`)
- otherwise, this function expression is independent. (e.g., `foo: function bar() {}`)

So I think better if it makes `countNode` variable or something like rather than it adds `Property` and `MethodDefinition` as entry points.
E.g., `const node = isEmbedded(funcNode) ? funcNode.parent : funcNode;`.

Reply:
Yep good points